pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,210 @@
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_('Private key')} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
+                    <Button variant='secondary'
+                        onClick={async () => {
+                            const key = await cockpit.script("wg genkey");
+                            setPrivateKey(key.trim());
+                        }}
+                    >
+                        Generate private key
+                    </Button>
+                </InputGroup>
+            </FormGroup>
+            <FormGroup label={_('Public key')}>
+                <ClipboardCopy isReadOnly>{publicKey}</ClipboardCopy>
+            </FormGroup>
+            <FormGroup label={_('Listen port')} fieldId={idPrefix + '-listen-port-input'}>
+                <TextInput id={idPrefix + '-listen-port-input'} placeholder='Random port selected when unspecified' type='number' value={listenPort} onChange={(_, val) => { setListenPort(val) }} />
+            </FormGroup>
+            <FormGroup label={_('Addresses')} fieldId={idPrefix + '-addresses-input'}>
+                <TextInput id={idPrefix + '-addresses-input'} value={addresses} onChange={(_, val) => { setAddresses(val) }} placeholder='E.g. 10.0.0.1/24' />
+            </FormGroup>
+            <FormFieldGroup
+                header={
+                    <FormFieldGroupHeader
+                        titleText={{ text: _("Peers") }}
+                        actions={
+                            <Button
+                                variant='secondary'
+                                onClick={() => setPeers(peers => [...peers, { publicKey: '', endpoint: '', allowedIps: '' }])}
+                            >
+                                {_("Add peer")}
+                            </Button>
+                        }
+                    />
+                }
+                className='dynamic-form-group'
+            >
+                {(peers.length !== 0)
+                    ? peers.map((peer, i) => (
+                        <Grid key={i} hasGutter>
+                            <FormGroup className='pf-m-6-col-on-md' label={_("Public key")} fieldId={idPrefix + '-publickey-peer-' + i}>
+                                <TextInput
+                                    value={peer.publicKey}
+                                    onChange={(_, val) => {
+                                        setPeers(peers => peers.map((peer, index) => i === index ? { ...peer, publicKey: val } : peer));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230703-110332-55dbfd0f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)